pallets/schemas/src/lib.rs

```diff
@@ -19,12 +24,16 @@ pub use pallet::*;
 pub mod weights;
 pub use weights::*;
 
+type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
 #[frame_support::pallet]
```

Comment:
This is generic BalanceType, at some point we will have to define our token system for enabling payments for such things

Reply:
We will have to implement currency and adaptors for custom tokens